src/renderer/components/Wallet/WalletTransactions/WalletTransactionsMultiSignature.js

```diff
@@ -65,6 +65,11 @@ export default {
         if (publicKey === WalletService.getPublicKeyFromWallet(this.wallet_fromRoute)) {
           this.$set(this, 'fetchedTransactions', response.transactions)
           this.totalCount = response.totalCount
+          if (this.totalCount > 0) {
```

Comment:
I think the check needs improving here to only include transactions that require the wallet's signature at that point in time. I would do something like this:

```
let hasPendingTransaction = false
for (const transaction of response.transactions) {
  if (TransactionService.needsWalletSignature(transaction, WalletService.getPublicKeyFromWallet(this.wallet_fromRoute))) {
    hasPendingTransaction = true
    break
  }
}

if (hasPendingTransaction) {
  this.$store.dispatch('transaction/setWaitingMultisignature', true)
} else {
  this.$store.dispatch('transaction/setWaitingMultisignature', false)
}
```

This is used in the TransactionShowMultiSignature.vue component

Reply:
Alex, I did some little improvements. This is a computed property. 

```js
isWaitingForSignature () {
  const transactions = this.fetchedTransactions
  const wallet = WalletService.getPublicKeyFromWallet(this.wallet_fromRoute)

  function checkTransactionForWaitingSignatures (transaction) {
    return TransactionService.needsWalletSignature(
      transaction,
      wallet
    )
  }

  return transactions.some(t => checkTransactionForWaitingSignatures(t))
}
```